File: micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/stateless/sync/completions/resources/MyResourceTemplates.java
package io.micronaut.mcp.server.stateless.sync.completions.resources;

import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.annotations.ResourceTemplate;
import jakarta.inject.Singleton;

@Requires(property = "spec.name", value = "StatelessSyncResourceCompletionsTest")
//tag::clazz[]
@Singleton
class  MyResourceTemplates {
    @ResourceTemplate(
        uriTemplate = "file:///home/user/documents/{fileName}",
        name = "userDocument",
        title = "User Document"
    )
    String ref(String fileName) {
        if (fileName.equals("report.pdf")) {
            return "Report PDF";
        } else if (fileName.equals("data.csv")) {
            return "Data CSV";
        } else if (fileName.equals("notes.txt")) {
            return "Notes TXT";
        }
        return "";
    }
}
//end::clazz[]

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/completions/resources/MyResourceTemplates.kt
package io.micronaut.mcp.docs.completions.resources

//tag::imports[]
import io.micronaut.context.annotation.Requires
import io.micronaut.mcp.annotations.ResourceTemplate
import jakarta.inject.Singleton

//end::imports[]
@Requires(property = "spec.name", value = "MyResourceCompletionsTest")
//tag::clazz[]
@Singleton
class MyResourceTemplates {
    @ResourceTemplate(
        uriTemplate = "file:///home/user/documents/{fileName}",
        name = "userDocument",
        title = "User Document"
    )
    fun ref(fileName: String): String {
        return when (fileName) {
            "report.pdf" -> "Report PDF"
            "data.csv" -> "Data CSV"
            "notes.txt" -> "Notes TXT"
            else -> ""
        }
    }
}
//end::clazz[]
